ISISComputingGroup/ibex_gui
# Ticket4255 update java dependencies

### Description of work

Update the IBEX GUI to java 11.

To test in eclipse:
- Download a recent eclipse or grab `2019-06` from the share at `\\isis\inst$\Kits$\CompGroup\ICP\Developer Tools`
- Download the latest **Java 11 hotspot** JDK from https://adoptopenjdk.net/?variant=openjdk11&jvmVariant=hotspot and install it
- In eclipse, import the project as usual (if you have an "old" import, you probably want to remove it and reimport)
- In eclipse, go to Window -> preferences -> java -> installed JREs and select the JDK directory you installed above
- Then in the same preferences dialog go to "compiler" and set "compliance level" to 11.
- Set target platform and run as usual (in theory)

To test via maven:
- Ensure you have a maven version >=3.6.0, but not 3.6.1 as this has a bug.
- Double check which maven version you are running using `mvn -v`
- Run build.bat. This will take care of getting the appropriate JDK and setting JAVA_HOME etc automatically.

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/4255

### Acceptance criteria

GUI builds and runs on java 11.

### Unit tests

Most unit tests unchanged and should still pass.

### System tests

n/a

### Documentation

https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Java-11-migration-notes

---

#### Code Review


### Final Steps
